examples/mnist/train.go

```diff
@@ -0,0 +1,285 @@
+ *	distributed under the License is distributed on an "AS IS" BASIS,
+ *	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *	See the License for the specific language governing permissions and
+ *	limitations under the License.
+ */
+
+package mnist
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/gomlx/gomlx/backends"
+	. "github.com/gomlx/gomlx/graph"
+	"github.com/gomlx/gomlx/graph/nanlogger"
+	"github.com/gomlx/gomlx/ml/context"
+	"github.com/gomlx/gomlx/ml/context/checkpoints"
+	"github.com/gomlx/gomlx/ml/data"
+	"github.com/gomlx/gomlx/ml/layers"
+	"github.com/gomlx/gomlx/ml/layers/activations"
+	"github.com/gomlx/gomlx/ml/layers/batchnorm"
+	"github.com/gomlx/gomlx/ml/layers/regularizers"
+	"github.com/gomlx/gomlx/ml/train"
+	"github.com/gomlx/gomlx/ml/train/losses"
+	"github.com/gomlx/gomlx/ml/train/metrics"
+	"github.com/gomlx/gomlx/ml/train/optimizers"
+	"github.com/gomlx/gomlx/ml/train/optimizers/cosineschedule"
+	"github.com/gomlx/gomlx/ui/commandline"
+	"github.com/gomlx/gomlx/ui/gonb/plotly"
+	"github.com/gomlx/gopjrt/dtypes"
+	"github.com/janpfeifer/must"
+)
+
+var excludeParams = []string{"data_dir", "train_steps", "num_checkpoints", "plots"}
+
+type ContextFn func(ctx *context.Context) *context.Context
+
+var Models = map[string]struct {
+	ctx   ContextFn
+	model train.ModelFn
+}{
+	"logistic": {CreateSoftMaxModelContext, LinearModelGraph},
+	"cnn":      {CreateCnnModelContext, CnnModelGraph},
+}
+
+var Losses = map[string]losses.LossFn{
+	"cross-entropy": losses.BinaryCrossentropyLogits,
+
+	"triplet": func(labels, predictions []*Node) (loss *Node) {
+		return losses.TripletLoss(labels, predictions, losses.TripletMiningStrategyAll, 1.0, losses.PairwiseDistanceMetricL2)
+	},
+	"triplet softmax": func(labels, predictions []*Node) (loss *Node) {
+		return losses.TripletLoss(labels, predictions, losses.TripletMiningStrategyAll, -1.0, losses.PairwiseDistanceMetricL2)
+	},
+	"triplet softmax with cosine distance": func(labels, predictions []*Node) (loss *Node) {
+		return losses.TripletLoss(labels, predictions, losses.TripletMiningStrategyAll, -1.0, losses.PairwiseDistanceMetricCosine)
+	},
+	"hard triple softmax": func(labels, predictions []*Node) (loss *Node) {
+		return losses.TripletLoss(labels, predictions, losses.TripletMiningStrategyHard, -1.0, losses.PairwiseDistanceMetricL2)
+	},
+	"semi-hard triplet softmax": func(labels, predictions []*Node) (loss *Node) {
+		return losses.TripletLoss(labels, predictions, losses.TripletMiningStrategySemiHard, -1.0, losses.PairwiseDistanceMetricL2)
+	},
+	"semi-hard triplet softmax with cosine distance": func(labels, predictions []*Node) (loss *Node) {
+		return losses.TripletLoss(labels, predictions, losses.TripletMiningStrategySemiHard, -1.0, losses.PairwiseDistanceMetricCosine)
+	},
+}
+
+// CreateCnnModelContext sets the context with default hyperparameters to use with TrainModel.
+func CreateSoftMaxModelContext(ctx *context.Context) *context.Context {
```

Comment:
Name mismatch (func and comment).

But, following the model.go, don't you want to call this CreateLinearModelContext ?